diag_manager/fms_diag_dlinked_list.F90

```diff
@@ -41,282 +41,297 @@
 !> @addtogroup fms_diag_dlinked_list_mod
 !> @{
 MODULE fms_diag_dlinked_list_mod
-  USE fms_mod, ONLY: error_mesg, FATAL, WARNING, NOTE
-  implicit none
-  !!TODO: COnsider setting the access (public,private) to functions, etc.
-  !> The doubly-linked list node type.
-  type, public:: FmsDlListNode_t
-    private
-    class(*), pointer :: data => null()        !< The data pointed to by the node.
-    type(FmsDlListNode_t), pointer :: next => null() !< A pointer to the previous node.
-    type(FmsDlListNode_t), pointer :: prev => null() !< A pointer to the next node.
-  end type FmsDlListNode_t
+   USE fms_mod, ONLY: error_mesg, FATAL, WARNING, NOTE
+   implicit none
+   !> The doubly-linked list node type.
+   type, public:: FmsDlListNode_t
+      private
+      class(*), pointer :: data => null()        !< The data pointed to by the node.
```

Comment:
[You can't name this variable `data`](https://github.com/NOAA-GFDL/FMS/blob/main/CODE_STYLE.md#general-1).  `DATA` is a reserved word in Fortran.

Reply:
Data used as a type or type qualifier has a different meaning than used here. For a member variable name, its OK. The program works with both Intel and gcc compilers. Also see "Modern Fortran Explained, Incorporating Fortran 2018", Metcalf, pg 317 vs pg 154.

Reply:
OK, I guess I phrased this incorrectly because, yes, Fortran *allows* you to name variables after reserved words.  However, our [FMS style guide does not allow it](https://github.com/NOAA-GFDL/FMS/blob/main/CODE_STYLE.md#general-1), and it specifically mentions `data` as a variable name that you shouldn't use in our code.  If you has a goto statement or a common black, I would ask you to remove them as well.

Reply:
OK. Changed!